### ISSUE TITLE
Can't log in on NMM

### ISSUE THREAD

**Issue by xAmbertje** [Nexus-Mods/Nexus-Mod-Manager#237]
Like the title says. Whenever I try to login, it gives me this error "Cannot reach the Nexus login server. Either your firewall is blocking NMM or the login server is down." I can however, log in on the website... This has been going on since yesterday morning.

**Comment by Arefu** [comment]
Did you check to make sure your Firewall allows NMM to connect to the internet?

**Comment by xAmbertje** [comment]
Yes I did, I went to check in Windows Defender and I allowed NMM to have acces to the internet. The thing is that I never had problems with NMM before, it worked fine at first. I didn't download anything either, it just stopped working one day.

**Comment by Arefu** [comment]
Have you tried the most recent version?

**Comment by curiousletterhead** [comment]
I'm having the same issue. Never happened before, allowed NMM in Windows Defender, and tried uninstalling and updating to the latest version. I still can't log in.

**Comment by Arefu** [comment]
Can you post log please? Pastebin preferred

**Comment by stormdrac** [comment]
sorry, I am having the same issue, I have had the same issue since I downloaded NMM. would you mind telling me what log? and where to get it? and I will happily post a log with Pastebin.

**Comment by Arefu** [comment]
Is in documents under nexus mod manage

**Comment by stormdrac** [comment]
i'm an old lady you gotta be a bit more specific, what is it labeled as? i see nothing in that folder that says "log". there is one under the steam/steamapps/common/fallout4/logs/ 
there two folders in that one labeled Bnet and Script. ? learning as i go here. thanks

**Comment by Arefu** [comment]
Sorry, the full path is:
C:\Users\<Username>\Documents\Nexus Mod Manager\Logs

Or very close to it as I'm on my phone

**Comment by stormdrac** [comment]
I am afraid I don't have anything called logs in my NMM ? idk i am at a loss sorry

**Comment by Arefu** [comment]
Hmm have a look around the folder them, I can't get near my computer at the moment

**Comment by Lucidi0us** [comment]
Having same issue on 0.64.3 version, but did not have it with 0.64.2.   Also, the 0.64.3 hasn't created any logs; the only logs I have are from 0.64.2 or older versions

Edit:  Correction, I thought I had 0.64.3 but had 0.64.2.   In 0.64.3 I can log in again, 0.64.2 was the problematic version for me.  Thanks!

**Comment by Lucidi0us** [comment]
OK, the first login that worked said it was sending a token.  On the next reload of NMM, it asked me to enter username and password and is giving me this error again (version is 0.64.3 for certain this time)

**Comment by Lucidi0us** [comment]
Enabled trace logging and it doesn't have any entries for login.  After NMM fully loads I don't see any entries beyond:

Setting up listener on Nexus Mod Manager-Fallout4IpcServer/Fallout4Listener at 2018-03-16 23:49:48
Running Application.

**Comment by Arefu** [comment]
I'm not reading this, put it on paste bin for the love of everything

**Comment by Lucidi0us** [comment]
There's no point in reading it at all.  No log entries in it relate to login to Nexus, they stop after the app is done finding mods and initializing.  I just deleted it since it seemed to bug you.  There is an issue with login, but the logs don't have any interesting data for it.

**Comment by Arefu** [comment]
Well then do you want to post your Firewall rules for allowing NMM through? Or specifically add it to the Firewall so it's allowed through explicitly.

**Comment by Lucidi0us** [comment]
Sure thing.

Here is my physical firewall's rule for outgoing connections (allow all)
![image](https://user-images.githubusercontent.com/2955403/37563052-a9f7bd4a-2a3d-11e8-9c40-0fc3d32bdfc1.png)

I have Bitdefender for the local, software firewall.  I have NMM whitelisted, and I can see it logging that it has let it pass:
![image](https://user-images.githubusercontent.com/2955403/37563066-f1dba5ea-2a3d-11e8-9fa6-73050d6f661f.png)

I have no issues logging in with versions <= 0.63.1

**Comment by Arefu** [comment]
For the sake of it do you want to check Window's firewall just to appease my attention to detail

**Comment by Lucidi0us** [comment]
Here you go:

![image](https://user-images.githubusercontent.com/2955403/37563085-6a833670-2a3e-11e8-9402-a1335d9510f4.png)

**Comment by Lucidi0us** [comment]
Did some poking around.  The log from Bitdefender must have been when I was using 0.63.1 in earlier testing.  Neither the physical firewall nor Bitdefender logs a connection attempt from 0.63.4 at all when I try to log in.  On 0.63.1, I see logs in both.

On 0.64.3, I can see in Resource Monitor that it is attempting to open a connection to 77.72.3.39.  I am still looking for anything else that might help.  I am able to ping, trace, and connect to an nginx on EPEL! page when using chrome with that IP.

![image](https://user-images.githubusercontent.com/2955403/37563161-2fda138e-2a40-11e8-9c38-65c5b3877942.png)

**Comment by Lucidi0us** [comment]
Something else for you.  If I run netstat -b, I do not see any data for NexusClient.exe when spamming logon attempts.  After sitting idle for about 2 minutes, one entry showed up in the netstat output.  The IP address corresponds to GitHub, so I am guessing it was probably checking for NNM updates at that time.  I am going to run Wireshark and attempt some more log-on's.

[NexusClient.exe]
  TCP    10.0.0.10:50349        151.101.48.133:https   ESTABLISHED

**Comment by Lucidi0us** [comment]
Checked with Wireshark and confirmed with Microsoft Message Alayzer, NMM 0.64.3 does not make any network connections when attempting to sign-in.  The only time it makes network connections appears to be when checking GitHub for a new version.

Here are the various versions from the tracelogs, in case there is something unique about my setup.

OS version: Microsoft Windows NT 6.2.9200.0
Installed .NET Versions:
	CDF SP 0
	v2.0.50727 SP 2
	v3.0 SP 2
	v3.5 SP 1
	v4 SP 0
	v4.0 SP 0
	v4.5: 4.6 or later

**Comment by Lucidi0us** [comment]
I haven't found any more data that seems useful.  For now, I have 0.64.3 and 0.64.1 installed to separate directories so I can continue to download mods, but still test 0.64.3 for you if you need more feedback.

**Comment by Lucidi0us** [comment]
This issue magically resolved itself recently.  I can log in on 0.64.3 now.  I haven't used my computer in a couple weeks, so I am not sure exactly when it started working.

**Comment by PeterHughes** [comment]
I'm having exactly the same issue as the original poster "Cannot reach the Nexus login server. Either your firewall is blocking NMM or the login server is down.". I've updated to Nexus.Mod.Manager-0.65.4. I've disabled Windows & Norton firewalls. Initially it prompts me to enter username & password. If I type a random username it says "Invalid Username or password". If I type the correct username or wrong or correct password, then it give the above "cannot reach..." error.

**Comment by Lucidi0us** [comment]
@PeterHughes This issue was impacting me 2 months ago and self-resolved.

TL;DR:  It's probably not your firewall if it you were able to login on previous versions.  No idea why it started to work for me, nothing had been recently changed to cause that.

I believe there is an issue either with the logon server or an ISP/interchange level issue.  Some in the forums have suggested that the login server gets DDoS'd from time to time, but it doesn't make sense if that is only impacting certain users.  ISP level issues would impact all users from the same ISP, so it could be that.  One of the really telling things I saw was that when using an older version of NMM, I could log in fine.  If they use the same server to login, then it doesn't make much sense that one version would work and the other wouldn't unless something outside of your system configuration was causing it.

**Comment by PeterHughes** [comment]
Fixed: I was using the wrong username. Not sure why that would return a "cannot reach server..." error, but it did...

**Comment by BoopleSnuffe** [comment]
Same problem; Can login just on the website, but NMM will not let me login.

**Comment by notnotbilly** [comment]
Installing the 0.65.2 version fixed it for me -- thanks for that!

**Comment by AgentRG** [comment]
I had 0.63 version installed and had the same issue. After I've updated to 0.65.2, the error went away and I was able to login again.

**Comment by squid-box** [comment]
This issue has several problems listed, and is getting pretty old, so grab [NMM 0.65.9](https://github.com/Nexus-Mods/Nexus-Mod-Manager/releases/tag/0.65.9) and see if you still have problems.

If you still can't log in with 0.65.9, create a new issue with a good description of what doesn't work, and include as much info as you can (trace log file if applicable).

Closing this issue as inactive/bloated :)

**Comment by bWildcard** [comment]
Adding both inbound and outbound connections in firewall and using your display name (not associated email) and password as allowed me to connect on 35.2

**Comment by kzzzzzzzzzzz** [comment]
last comment may be 2 years old but for the sake of those still using nmm or coming back to an old install, there is a new version of nexus mod manager still maintained and on nexus if you just search "nexus mod manager". still logs in and updates. installs like an update. theres a verify process once you run it you have to have been logged into nexus and few clicks and done.
